compiler.js

```diff
@@ -3,6 +3,62 @@ function isWholeProgramTarget(compileTarget)
     return compileTarget == "METAL" || compileTarget == "SPIRV";
 }
 
+const imageMainSource = `
+import user;
+
+uniform float time;
+RWStructuredBuffer<int>               outputBuffer;
+[format("r32f")] RWTexture2D<float>   texture;
+
+inline float encodeColor(float4 color)
+{
+    uint4 colorInt = { uint(color.x * 255.0f),
+                       uint(color.y * 255.0f),
+                       uint(color.z * 255.0f),
+                       uint(color.w * 255.0f) };
+
+    float encodedColor = float(colorInt.x << 24 | colorInt.y << 16 | colorInt.z << 8 | colorInt.w);
+    return encodedColor;
+}
+
+[shader("compute")]
+void imageMain(uint3 dispatchThreadID : SV_DispatchThreadID)
```

Comment:
area of impovement: let's use `[numthreads(16,16,1)]` here and add a bounds check before calling `imageMain`.

Reply:
yes, follow-up